tests/KubernetesClientConfigurationTests.cs

```diff
@@ -322,6 +324,76 @@ public void AsUserExtra()
 
             var cfg = KubernetesClientConfiguration.BuildConfigFromConfigFile(txt, null, null);
             Assert.NotNull(cfg.Host);
+        }
+
+        /// <summary>
+        ///     Ensures Kube config file is loaded from explicit file
+        /// </summary>
+        [Fact]
+        public void LoadKubeConfigExplicitFilePath()
+        {
+            var txt = File.ReadAllText("assets/kubeconfig.yml");
+            var expectedCfg = Yaml.LoadFromString<K8SConfiguration>(txt);
+
+            var cfg = KubernetesClientConfiguration.LoadKubeConfig("assets/kubeconfig.yml");
+
+            Assert.NotNull(cfg);
+
+            ConfigsEqual(expectedCfg, cfg);
+        }
+
+        private void ConfigsEqual(K8SConfiguration expected, K8SConfiguration actual)
```

Comment:
This function is too long, I think. Factor it so there is an

`assertContextEqual(...)`

`assertUserEqual(...)`

`assertClusterEqual(...)`

set of methods and refactor the for loops to just call those methods?